protocols/kad/src/kbucket.rs

```diff
@@ -120,6 +121,26 @@ impl KBucketsPeerId<PeerId> for Multihash {
     }
 }
 
+impl KBucketsPeerId for KadHash {
+    fn distance_with(&self, other: &Self) -> u32 {
+        <Multihash as KBucketsPeerId<Multihash>>::distance_with(self.hash(), other.hash())
+    }
+
+    fn max_distance() -> NonZeroUsize {
+        <Multihash as KBucketsPeerId>::max_distance()
```

Comment:
If we change the hash to a `[u8; 32]`, then that should be 256, and `distance_with` should be adjusted accordingly.

Reply:
I have changed this as well. I think we could abstract the xor metric away, but I'm missing a trait for `.leading_zeros()` in `U512` and `U256`. What do you think?

Reply:
Also for the case where we compare `KadHash` and `Multihash` I have taken `max_distance` to be 512 and converted both hashes to `U512`.

Reply:
I think we should split `distance_with` and `max_distance` in two different traits, something like:

```rust
trait MaxDistance: DistanceWith<Self> {
    fn max_distance() -> u32;
}

trait DistanceWith<TOther> {
    fn distance_with(&self, other: &TOther) -> u32;
}

Reply:
I would make that a new PR